xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/docextra.vm

```diff
@@ -55,6 +57,8 @@
       #set ($sortAttachmentsBy = "&sortAttachmentsBy=$escapetool.url(${sortAttachmentsBy})")
     #end
     #set ($discard = $docextras.add(['Attachments', 'attachments', $services.localization.render('docextra.attachments'), $doc.getAttachmentList().size(), "attachmentsinline.vm$!{sortAttachmentsBy}", $services.localization.render('core.shortcuts.view.attachments')]))
+    #getRequiredSkinExtensions($requiredSkinExtensions)
+    #set ($discard = $response.setHeader('X-XWIKI-HTML-HEAD', $requiredSkinExtensions))
```

Comment:
I don't think this is the right place. ``docextra.vm`` is evaluated on the initial HTML request, so on page load. This needs to be put where the content is loaded asynchronously, after the initial page load, probably in ``attachmentsinline.vm``.

Initially it was hard / complex to load JavaScript / CSS from a content that is fetched and injected dynamically after the page load. The solution was to preload the JavaScript / CSS on page load, which is what ``docextra.vm`` does here: it preloads the resources needed by the Attachments tab (``attachmentsinline.vm``). The downside is that we're loading resources that may not be needed by the user (if the user doesn't open the Attachments tab), thus increasing the page load time. It's better to defer the loading until those resources are needed, i.e. until the Attachments tab is loaded. So the solution is to move the resource includes from ``docextra.vm`` to ``attachmentsinline.vm`` (or some nested template) and to use the ``X-XWIKI-HTML-HEAD`` to load them when the Attachments tab is fetched asynchronously.

Reply:
This is already done in `attachmentsinline.vm` https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/attachmentsinline.vm#L22
Only that if I refresh the page while on the attachments tab, the resources are not loaded if they are not defined also inside  `doextra.vm`

Reply:
What I can see while debugging is that here https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/xwiki.js#L1862 the `getHeader('X-XWIKI-HTML-HEAD')` will not contain the required extensions, as if https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/attachmentsinline.vm#L60 was not done (even if `$requiredSkinExtensions` wasn't empty)